include/envoy/upstream/cluster_manager.h

```diff
@@ -227,6 +227,10 @@ class ClusterManager {
    * @return Config::SubscriptionFactory& the subscription factory.
    */
   virtual Config::SubscriptionFactory& subscriptionFactory() PURE;
+
+  virtual void updateClusterInterest(const std::set<std::string>& update_to_these_names) PURE;
+
+  virtual void addToClusterInterest(const std::set<std::string>& add_these_names) PURE;
```

Comment:
where is this used?

Reply:
well, `updateClusterInterest` here only provides the possibility to update the entire clusterInterest, maybe I should remove it. As for `addToClusterInterest`,  the code in rocketmq_proxy (#9503 ) call this method  in practice.

code in `RouterImpl#sendRequestToUpstream`
```cpp
  if (!cluster) {
    active_message_->awaitCluster(true);
    active_message_->incPendingStats();
    if (active_message_->connectionManager().hasInFlightCdsRequest(cluster_name)) {
      ENVOY_LOG(trace, "Cluster {} is absent, but there has been an in-flight CDS request",
                cluster_name);
      return;
    }

    std::set<std::string> resources = {cluster_name};
    cluster_manager_.addToClusterInterest(resources);
    active_message_->connectionManager().markInFlightCdsRequest(cluster_name);
    ENVOY_LOG(info, "Cluster {} is not available, trigger to set up through delta CDS. Opaque: {}",
              cluster_name, opaque);
    return;
  }
```

code in `ActiveMessage#onQueryTopicRoute`
```cpp
    await_cluster_ = true;
    incPendingStats();
    std::set<std::string> resources{cluster_name};
    ENVOY_LOG(trace, "Initiate on-demand cluster discovery service for {}", cluster_name);
    clusterManager.addToClusterInterest(resources);
    connection_manager_.markInFlightCdsRequest(cluster_name);
```

Reply:
@wgallagher the on-demand part in #9503 has been removed temporarily, but you can still view the related part in this commit : https://github.com/aaron-ai/envoy/commit/40ed8368b8d61e847d21f6c37b894d8a74dd0efb